wallet/test/Test/Pos/Wallet/Web/Pending/SubmissionSpec.hs

```diff
@@ -0,0 +1,194 @@
+module Test.Pos.Wallet.Web.Pending.SubmissionSpec
+       ( spec
+       ) where
+
+import           Universum
+
+import           Data.Default (def)
+import           Test.Hspec (Spec, describe)
+import           Test.Hspec.QuickCheck (modifyMaxSuccess)
+import           Test.QuickCheck (Gen, arbitrary, generate)
+import           Test.QuickCheck.Arbitrary.Generic (genericArbitrary)
+import           Test.QuickCheck.Monadic (PropertyM)
+
+import           Pos.Client.Txp.History (TxHistoryEntry (..))
+import           Pos.Core (Timestamp (..), addMicrosecondsToTimestamp)
+import           Pos.Launcher (HasConfigurations)
+import           Pos.Txp.Toil.Failure (ToilVerFailure (..))
+import           Pos.Util.CompileInfo (HasCompileInfo, withCompileInfo)
+import           Pos.Wallet.Web.ClientTypes (CHash (..), CId (..))
+
+import           Pos.Wallet.Web.Pending.Submission (TxSubmissionMode, TxSubmissionResult (..),
+                                                    submitAndSavePtxStates)
+import           Pos.Wallet.Web.Pending.Types (PendingTx (..), PtxCondition (..), PtxPoolInfo)
+import           Pos.Util.QuickCheck.Property (assertProperty)
+import           Test.Pos.Configuration (withDefConfigurations)
+import           Test.Pos.Wallet.Web.Mode (walletPropertySpec)
+
+
+-- stack test cardano-sl-wallet --fast --test-arguments "-m Wallet.Web.Pending.SubmissionSpec"
+spec :: Spec
+spec = withCompileInfo def $
+       withDefConfigurations $
+    describe "Wallet.Web.Pending.SubmissionSpec" $ modifyMaxSuccess (const 100) $ do
+        describe "submitAndSaveTx" $ do
+            describe "Normal application" normalApplicationSpec
+            describe "Tx timeout after an hour " txTimeoutSpec
+            describe "ToilKnown exception occurs " toilKnownLargeSpec
+            describe "ToilTooLarge exception occurs " toilTooLargeSpec
+
+-- testTime :: UTCTime
+-- testTime = parseTime True defaultTimeLocale "%Y-%m-%dT%H:%M:%S%z" "2017-03-29T11:02:57+00:00 "
+
+
+normalApplicationSpec :: (HasCompileInfo, HasConfigurations) => Spec
+normalApplicationSpec = walletPropertySpec decription $ do
+    let timestamp = Timestamp $ 1518968949 * 1000000 -- In microseconds
```

Comment:
As far s I understand, actual time doesn't matter.
Would be nice to set just `0` here or extract this `timestamp` somewhere in order not to call the eyes

Reply:
Yes, actual time doesn't matter, but a real value adds value to the test since the user can have an example of how the code actually works, and we test a more realistic scenario.